### PR TITLE
[tools] Add diagnostics when a native library turns off -dead_link support.

### PR DESCRIPTION
**Pull request by rolfbjarne**

### CODE REVIEW

File: tools/common/Assembly.cs [hunk 1]
```diff
@@ -307,8 +307,10 @@ void AssertiOSVersionSupportsUserFrameworks (string path)
 		void ProcessNativeReferenceOptions (NativeReferenceMetadata metadata)
 		{
 			// We can't add -dead_strip if there are any LinkWith attributes where smart linking is disabled.
-			if (!metadata.SmartLink)
+			if (!metadata.SmartLink) {
+				Driver.Log (3, $"The library '{metadata.LibraryName}', shipped with the assembly '{FullName}', sets SmartLink=false, which will disable passing -dead_strip to the native linker (and make the app bigger).")
```
Comment (mandel-macaque) [inline tools/common/Assembly.cs:311]:
```suggestion
				Driver.Log (3, $"The library '{metadata.LibraryName}', shipped with the assembly '{FullName}', sets SmartLink=false, which will disable passing -dead_strip to the native linker (and make the app bigger).");
```

**Review by mandel-macaque — CHANGES_REQUESTED** [review]
; missing

**Review by mandel-macaque — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Comment by rolfbjarne** [comment]
Test failures are unrelated (https://github.com/xamarin/maccore/issues/2450).

**Pull request merged**